### PR TITLE
Several improvements

### PR DESCRIPTION
**Pull request by mrdrogdrog**
### Description
This PR improves the development toolchain in the repository by:
- Adding Jetbrains Copyright configuration
- Adding semver for dependencies (it's a lib)
- Removing the license information from the PR template file (it's annoying to have this in github when opening a PR)
- Adding prettiert to eslint
- Moving the prettier config to a separate file
- Adding the correct license for yarn

### Steps

- [x] Added changes
- [x] I read the [contribution documentation](https://github.com/hedgedoc/html-to-react/blob/main/CONTRIBUTING.md) and signed-off my commits to accept the DCO.

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -1,18 +1,19 @@
-# SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
+# SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
```
Comment (DerMolly) [inline .github/workflows/lint.yml:1]:
```suggestion
# SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
```

File: .github/workflows/reuse.yml [hunk 2]
```diff
@@ -1,10 +1,14 @@
-# SPDX-FileCopyrightText: 2020 The HedgeDoc developers (see AUTHORS file)
+# SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
```
Comment (DerMolly) [inline .github/workflows/reuse.yml:1]:
```suggestion
# SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
```

File: .prettierignore.license [hunk 3]
```diff
@@ -0,0 +1,4 @@
+SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
```
Comment (DerMolly) [inline .prettierignore.license:1]:
```suggestion
SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
```

File: .prettierrc.json.license [hunk 4]
```diff
@@ -0,0 +1,3 @@
+SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
```
Comment (DerMolly) [inline .prettierrc.json.license:1]:
```suggestion
SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
```

File: .reuse/dep5 [hunk 5]
```diff
@@ -3,6 +3,14 @@ Upstream-Name: html-to-react
...
+Copyright: 2021 The HedgeDoc developers (see AUTHORS file)
+License: CC-BY-SA-4.0
+
 Files: .yarn/**
 Copyright: Yarn contributors
-License: BSD-2-Clause
+License: LicenseRef-yarn
+
+Files: .idea/**
+Copyright: 2021 The HedgeDoc developers (see AUTHORS file)
```
Comment (DerMolly) [inline .reuse/dep5:15]:
```suggestion
Copyright: 2022 The HedgeDoc developers (see AUTHORS file)
```

File: .reuse/dep5 [hunk 6]
```diff
@@ -3,6 +3,14 @@ Upstream-Name: html-to-react
 Upstream-Contact: The HedgeDoc developers <license@hedgedoc.org>
 Source: https://github.com/hedgedoc/html-to-react
 
+Files: .github/pull_request_template.md
+Copyright: 2021 The HedgeDoc developers (see AUTHORS file)
```
Comment (DerMolly) [inline .reuse/dep5:7]:
```suggestion
Copyright: 2022 The HedgeDoc developers (see AUTHORS file)
```

File: package.json [hunk 7]
```diff
@@ -7,10 +7,9 @@
   "scripts": {
     "build": "tsc-esm && tsc -p tsconfig-cjs.json && find dist/cjs -type d -execdir sh -c 'cd {}; mmv -rd \"*.js*\" \"#1#2\"' \\;",
     "test": "jest",
-    "prepublish": "rm -rf dist && yarn format && yarn lint && yarn build && yarn test",
+    "prepublish": "rm -rf dist && yarn lint && yarn build && yarn test",
     "lint": "eslint src --ext .ts",
-    "format": "prettier -c \"src/**/*.{ts,tsx,js}\"",
-    "format:fix": "prettier -w \"src/**/*.{ts,tsx,js}\""
+    "lint:fix": "eslint --fix --ext .ts src"
```
Comment (DerMolly) [inline package.json:12]: `format`?

Comment (mrdrogdrog) [reply package.json:12]: I added the eslint plugin for prettier. Therefore lint checks the code format using prettier and lint:fix reformats the code.

**Review by DerMolly — APPROVED** [review]

**Pull request merged**